src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -908,8 +909,15 @@ export class TabsTitleControl extends TitleControl {
 	private redrawEditorDirty(editor: IEditorInput, tabContainer: HTMLElement): void {
 		if (editor.isDirty()) {
 			addClass(tabContainer, 'dirty');
+
+			if (this.accessor.partOptions.dirtyTabBorder && !this.getColor(TAB_ACTIVE_BORDER_TOP)) {
```

Comment:
Sorry for not being clear here: instead of assigning a CSS class, why not set the border color directly on the element? I wanted to avoid having a new CSS rule produced for this feature.